smart_card_connector_app/src/window-devices-displaying.js

```diff
@@ -55,26 +55,32 @@ var readersListElement = goog.dom.getElement('readers-list');
  */
 var addDeviceElement = goog.dom.getElement('add-device');
 
-function loadDeviceList() {
-  logger.fine('Requesting available USB devices list...');
-  chrome.usb.getDevices({'filters': USB_DEVICE_FILTERS}, getDevicesCallback);
-}
-
 /**
- * @param {!chrome.usb.Device} device
+ * @type {GSC.PcscLiteServer.ReaderTracker}
  */
-function usbDeviceAddedListener(device) {
-  logger.fine('A USB device was added: ' + makeReaderDescriptionString(device));
-  loadDeviceList();
+var readerTracker;
+
+function displayReaderList(readers) {
+  logger.info(readers.length + ' card reader(s) available: ' +
+              GSC.DebugDump.dump(readers));
+  goog.dom.removeChildren(readersListElement);
+
+  for (let reader of readers) {
+    GSC.Logging.checkWithLogger(logger, !goog.isNull(readersListElement));
+    goog.asserts.assert(readersListElement);
+    // TODO(isandrk): Put the error in a new line so it's copyable.
+    var errorIndicator = goog.dom.createDom(
+        'div', 'error-indicator color-' + reader['status']);
+    var title = reader['error'] ? {title: 'Reader error, code = ' +
+                                   reader['error']} : undefined;
+    var element = goog.dom.createDom(
+        'li', title, errorIndicator, reader['name']);
+    goog.dom.append(readersListElement, element);
+  };
 }
 
-/**
- * @param {!chrome.usb.Device} device
- */
-function usbDeviceRemovedListener(device) {
-  logger.fine(
-      'A USB device was removed: ' + makeReaderDescriptionString(device));
-  loadDeviceList();
+function readerUpdatedListener(readers) {
```

Comment:
nit: You may just remove this function, it's basically a synonym to displayReaderList.